File: gluecodium/src/test/resources/smoke/external_types/output/android/com/example/dontsmoke/SerializableStructWithExternalField.java
/*
 *
 */
package com.example.dontsmoke;
import android.os.Parcel;
import android.os.Parcelable;
import android.support.annotation.NonNull;
public final class SerializableStructWithExternalField implements Parcelable {
    public static final Parcelable.Creator<SerializableStructWithExternalField> CREATOR = new Parcelable.Creator<SerializableStructWithExternalField>() {
        public SerializableStructWithExternalField createFromParcel(final Parcel in_parcel) {
            return new SerializableStructWithExternalField(in_parcel);
        }
        public SerializableStructWithExternalField[] newArray(final int size) {
            return new SerializableStructWithExternalField[size];
        }
    };
    @NonNull
    public com.here.android.test.AnExternalStruct someStruct;
    public SerializableStructWithExternalField(@NonNull final com.here.android.test.AnExternalStruct someStruct) {
        this.someStruct = someStruct;
    }
    private SerializableStructWithExternalField(final Parcel in_parcel) {
        this.someStruct = in_parcel.readParcelable(Thread.currentThread().getContextClassLoader());
    }
    @Override
    public int describeContents() {
        return 0;
    }
    @Override
    public void writeToParcel(final Parcel out_parcel, final int flags) {
        out_parcel.writeParcelable(someStruct, 0);
    }
}

File: gluecodium/src/test/resources/smoke/external_types/output/android-kotlin/com/example/kotlinsmoke/SerializableStructWithExternalField.kt
/*

 *
 */

@file:JvmName("SerializableStructWithExternalFieldExtensions")

package com.example.kotlinsmoke

import android.os.Parcel
import android.os.Parcelable

class SerializableStructWithExternalField : Parcelable {
    @JvmField var someStruct: com.here.android.test.AnExternalStruct



    constructor(someStruct: com.here.android.test.AnExternalStruct) {
        this.someStruct = someStruct
    }


    private constructor(parcel: Parcel) {
        this.someStruct = parcel.readParcelable(Thread.currentThread().getContextClassLoader())!!
    }

    override fun describeContents() = 0

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeParcelable(someStruct, 0)
    }





    companion object {
        @JvmField
        val CREATOR = object : Parcelable.Creator<SerializableStructWithExternalField> {
            override fun createFromParcel(parcel: Parcel) = SerializableStructWithExternalField(parcel)
            override fun newArray(size: Int) = arrayOfNulls<SerializableStructWithExternalField?>(size)
        }

    }
}

